clang/lib/CIR/Dialect/IR/CIRAttrs.cpp

```diff
@@ -155,6 +155,43 @@ void LangAttr::print(AsmPrinter &printer) const {
   printer << "<" << getLang() << '>';
 }
 
+//===----------------------------------------------------------------------===//
+// ConstPtrAttr definitions
+//===----------------------------------------------------------------------===//
+
+Attribute ConstPtrAttr::parse(AsmParser &parser, Type odsType) {
+  uint64_t value;
+
+  if (!odsType.isa<cir::PointerType>())
+    return {};
+
```

Comment:
Alternatively, you could introduce a `custom` directive on tablegen and customize only the constant printing: https://mlir.llvm.org/docs/DefiningDialects/AttributesAndTypes/#custom-and-ref-directive

Reply:
Never knew that, thanks for pointing it out! Sounds like we still need to provide the parser/printer functions, `printStructMembers` for example. I can give it a shot in future PRs.